### PR TITLE
Fix/fake parents

### PR DESCRIPTION
**Pull request by sashokbg**
I have opened this PR in place of #17.

As you have suggested the library now detects automatically if a parent node is a "real node" and it prevents its deletion.
To make this work I have added a class called "cdnd-fake-parent" for parents that can be removed.

I have also added E2E tests that prove it works.

You can run them with **npm run test:e2e** and check the generated report in **test/reports**

### CODE REVIEW

**Comment by maxkfranz** [comment]
I've reviewed the PR, and it looks great.

Before it's merged in, do you think it would make sense to use a class name like `cdnd-new-parent` to be consistent with the naming of `newParentNode()` in `options`?

https://github.com/cytoscape/cytoscape.js-compound-drag-and-drop#api

**Comment by sashokbg** [comment]
Hello @maxkfranz I have updated the code to change the class to cdnd-new-parent and also rebased the branch with master.

File: README.md [hunk 1]
```diff
@@ -85,11 +85,11 @@ const options = {
   grabbedNode: node => true, // filter function to specify which nodes are valid to grab and drop into other nodes
   dropTarget: (dropTarget, grabbedNode) => true, // filter function to specify which parent nodes are valid drop targets
   dropSibling: (dropSibling, grabbedNode) => true, // filter function to specify which orphan nodes are valid drop siblings
-  newParentNode: (grabbedNode, dropSibling) => ({}), // specifies element json for parent nodes added by dropping an orphan node on another orphan (a drop sibling)
   boundingBoxOptions: { // same as https://js.cytoscape.org/#eles.boundingBox, used when calculating if one node is dragged over another
     includeOverlays: false,
     includeLabels: true
   },
+  newParentNode: (grabbedNode, dropSibling) => ({}), // specifies element json for parent nodes added by dropping an orphan node on another orphan (a drop sibling). You can chose to return the dropSibling in which case it becomes the parent node and will be preserved after all its children are removed.
```
Comment (maxkfranz) [inline README.md]: It looks like this got moved in the merge.  It should be beside the other functions, like dropSibling

**Review by maxkfranz — CHANGES_REQUESTED** [review]
Looks good.  I think the readme documentation just needs a few tweaks, and then the PR can be merged in.

Is everything new from this PR documented in the readme?  I don't see any additions for the new class name, for example.

**Comment by sashokbg** [comment]
Hello, I have documented the new class and also moved the callback description up. Sorry for the delay

**Review by philwronski — APPROVED** [review]

**Review by maxkfranz — APPROVED** [review]

**Comment by maxkfranz** [comment]
v1.1.0 released

**Pull request merged**